web/src/rows.ts

```diff
@@ -70,38 +74,32 @@ export function visible_range(start_id, end_id) {
     return rows;
 }
 
-export function is_overlay_row($row) {
+export function is_overlay_row($row: JQuery): boolean {
     return $row.closest(".overlay-message-row").length >= 1;
 }
 
-export function id($message_row) {
+export function id($message_row: JQuery): number | undefined {
     if (is_overlay_row($message_row)) {
         blueslip.error("Drafts and scheduled messages have no zid");
         return undefined;
     }
 
-    /*
-        For blueslip errors, don't return early, since
-        we may have some code now that actually relies
-        on the NaN behavior here.  We can try to clean
-        that up in the future, but we mainly just want
-        more data now.
-    */
-
     if ($message_row.length !== 1) {
         blueslip.error("Caller should pass in a single row.");
+        throw new Error("Caller should pass in a single row.");
```

Comment:
The `blueslip.error` is redundant. Just `throw` (or `assert`).

Reply:
Done